src/System.IO.Ports/src/System/IO/Ports/SerialStream.cs

```diff
@@ -1681,9 +1681,26 @@ internal bool ShutdownLoop
                 }
             }
 
+            /// <summary>
+            /// Call WaitForCommEvent (which is a thread function for a background thread)
+            /// within an exception handler, so that unhandled exceptions in WaitForCommEvent
+            /// don't cause process termination
+            /// </summary>
+            internal void SafelyWaitForCommEvent()
+            {
+                try
+                {
+                    WaitForCommEvent();
+                }
+                catch(Exception ex)
```

Comment:
@danmosemsft I put in an Debug.Fail for every exception following your suggestion yesterday - is that enough?